src/UpupUploader.tsx

```diff
@@ -60,6 +63,13 @@ export type UploadFilesRef = {
     dynamicUploadFiles: (files: File[]) => Promise<string[] | null>
 }
 
+// Add this helper function at the top level
+const createFileWithId = (file: File) => {
+    return Object.assign(file, {
+        id: `${file.name}-${file.size}-${file.lastModified}-${Math.random()}`,
```

Comment:
Using Math.random() for unique IDs can still lead to collisions. I know the chance is small but replacing it with uuid would ensure reliability.

Reply:
@BSalaeddin Thanks man, done